python/lsst/ap/association/ssoAssociation.py

```diff
@@ -0,0 +1,131 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""Spatial association for Solar System Objects."""
+
+__all__ = ["SolarSystemAssociationConfig", "SolarSystemAssociationTask"]
+
+import numpy as np
+from scipy.spatial import cKDTree
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+
+class SolarSystemAssociationConfig(pexConfig.Config):
+    """Config class for SolarSystemAssociationTask.
+    """
+    maxDistArcSeconds = pexConfig.Field(
+        dtype=float,
+        doc='Maximum distance in arcseconds to test for a DIASource to be a '
+        'match to a SSObject.',
+        default=2.0,
+    )
+
+
+class SolarSystemAssociationTask(pipeBase.Task):
+    """Associate DIASources into existing SolarSystem Objects.
+
+    This task performs the association of detected DIASources in a visit
+    with known solar system objects.
+    """
+    ConfigClass = SolarSystemAssociationConfig
+    _DefaultName = "association"
+
+    @pipeBase.timeMethod
+    def run(self, diaSourceCatalog, solarSystemObjects):
+        """Create a searchable tree of unassociated DiaSources and match
+        to the nearest ssoObject.
+
+        Parameters
+        ----------
+        diaSourceCatalog : `pandas.DataFrame`
+            Catalog of DiaSources. Modified in place to add ssObjectId to
+            successfully associated DiaSources.
+        solarSystemObjects : `pandas.DataFrame`
+            Set of solar system objects that should be within the footprint
+            of the current visit.
+
+        Returns
+        -------
+        resultsStruct : `lsst.pipe.base.Struct`
+
+            - ``ssoAssocDiaSources`` : Set of DiaSources associated with
+              solar system objects in this visit. (`pandas.DataFrame`)
+            - ``unAssocDiaSources`` : Set of DiaSources that were unassociated
+              with any solar system object. (`pandas.DataFrame`)
+        """
+        maxRadius = np.deg2rad(self.config.maxDistArcSeconds / 3600)
+
+        # Transform DIA RADEC coordinates to unit sphere xyz for tree building.
+        vectors = self._radec_to_xyz(diaSourceCatalog["ra"],
+                                     diaSourceCatalog["decl"])
+
+        # Create KDTree of DIA sources
+        tree = cKDTree(vectors)
+
+        # Query the KDtree for DIA nearest neighbors to SSOs. Currently only
+        # picks the DiaSource with the shortest distance. We can do something
+        # fancier later.
+        for index, ssObject in solarSystemObjects.iterrows():
+
+            # convert SSO radec to ICRF position
+            ssoVect = self._radec_to_xyz(ssObject["ra"], ssObject["decl"])
```

Comment:
This isn't converting to ICRF. I don't think a comment is necessary here anyway, since the method name is clear.

Reply:
Left over comment from a previous version. Now removed.